binderhub/static/js/index.js

```diff
@@ -231,19 +228,35 @@ function build(providerSpec, log, path, pathType) {
 }
 
 function setUpLog() {
-  var log = new Terminal({
+  const log = new Terminal({
     convertEol: true,
     disableStdin: true
   });
+  const fitAddon = new FitAddon();
+  log.loadAddon(fitAddon);
+
+  var $logContainer = $("div#log-container");
+  var $panelBody = $("div.panel-body");
+  log.fit = function () {
+    if (!$panelBody.hasClass('hidden')) {
+      fitAddon.fit();
+    }
+  };
 
-  log.open(document.getElementById('log'), false);
+  // https://xtermjs.org/docs/api/terminal/classes/terminal/#open says that
+  // `parent` must be visible (have dimensions) when open is called 
+  // as several DOM- based measurements need to be performed when this function is called.
+  $panelBody.removeClass('hidden');
```

Comment:
Because `open` method needs the parent elements to be visible, here I made them visible, then called the open method and then made them hidden again. How can we improve this?